src/LayerEditor/ui/data/le_decomposition.py

```diff
@@ -53,42 +70,36 @@ def new_point(self, pos, last_point):
     def initialize_regions(self):
         """Initializes regions for all shapes which dont have defined regions. If region is defined it is ignored."""
         for dim in range(3):
-            for item_id in self.decomp.shapes[dim]:
+            for item_id in self.poly_decomp.decomp.shapes[dim]:
                 self.block.init_regions_for_new_shape(dim, item_id)
 
     def delete_items(self, items):
         """Delete all points and segments"""
         with undo.group("Delete selected"):
             for dim, id in items:
                 if dim == 1:
-                    self.delete_segment(self.segments[id])
+                    self.poly_decomp.delete_segment(self.poly_decomp.segments[id])
 
             for dim, id in items:
                 if dim == 0:
-                    self.delete_point(self.points[id])
+                    self.poly_decomp.delete_point(self.poly_decomp.points[id])
 
     def copy_itself(self):
         """ Make new decomposition with identical topology and nodeset as this.
             Ids will change, so everything that uses shape ids needs to be updated using old_to_new_id"""
-        receiver = undo.stack()._receiver
+        receiver = undo.stack()._receiver   # bgem changes the receiver which is not desired this fixes it
```

Comment:
Jestli je pointa vypnout behem kopie UNDOO mechanizkums, neni lepsi pouzit 
context manager, jako jinde?

Reply:
`with undo.pause_undo()`

Reply:
Nejde o vypnutí undo, ale o vrácení receiveru na původní objekt. Při různých situacích se v bgemu mění nejen obsah objektu v _receiver, ale i samotný objekt. Děje se to například při groupování akcí. Zde mi bgem změníl objekt v _reciver a pravděpodobně způsobilo že místo některé akce v groupě chyběli, protože byly v bgemu zahozeny. Nicméně si přesně nevzpomínám jaký problém to způsobilo, pouze hádám.